### PR TITLE
Add some entity mappings, more general name for EntityParts

### PR DESCRIPTION
**Pull request by mezz**

### CODE REVIEW

**Review by Darkhax — APPROVED** [review]

File: mappings/net/minecraft/entity/Entity.mapping [hunk 1]
```diff
@@ -322,6 +329,8 @@ CLASS none/sg net/minecraft/entity/Entity
 		ARG 0 entity
 	METHOD r isTeammate (Lnone/sg;)Z
 		ARG 0 entity
+	METHOD s isEqualEntity (Lnone/sg;)Z
```
Comment (asiekierka) [inline mappings/net/minecraft/entity/Entity.mapping]: Why not just isEqual? Or isEqualType? Not sure what it does...

File: mappings/net/minecraft/entity/decoration/EntityDecorationBase.mapping [hunk 2]
```diff
@@ -23,6 +23,7 @@ CLASS none/yo net/minecraft/entity/decoration/EntityDecorationBase
 		ARG 0 entity
 	METHOD a onStruckByLightning (Lnone/yx;)V
 		ARG 0 lightning
+	METHOD aq hasCollision ()Z
```
Comment (asiekierka) [inline mappings/net/minecraft/entity/decoration/EntityDecorationBase.mapping]: doesCollide? canCollide?

File: mappings/net/minecraft/entity/Entity.mapping (hunk 1 above)
Comment (mezz) [reply mappings/net/minecraft/entity/Entity.mapping]: The param type here is `Entity`, so I don't want to override `isEqual`.
It returns true if the param == this, or if this is an entity part and the param is the entity that owns that part.

File: mappings/net/minecraft/entity/decoration/EntityDecorationBase.mapping (hunk 2 above)
Comment (mezz) [reply mappings/net/minecraft/entity/decoration/EntityDecorationBase.mapping]: hmm I agree it's not quite right but I don't know what's best.

File: mappings/net/minecraft/entity/Entity.mapping (hunk 1 above)
Comment (asiekierka) [reply mappings/net/minecraft/entity/Entity.mapping]: hasEqualParts?

Comment (mezz) [reply mappings/net/minecraft/entity/Entity.mapping]: sounds good

Comment (mezz) [reply mappings/net/minecraft/entity/Entity.mapping]: well, not quite actually.
`dragonPart.isEqualEntity(dragon)`

File: mappings/net/minecraft/entity/decoration/EntityDecorationBase.mapping (hunk 2 above)
Comment (mezz) [reply mappings/net/minecraft/entity/decoration/EntityDecorationBase.mapping]: I'll go with doesCollide

File: mappings/net/minecraft/entity/Entity.mapping (hunk 1 above)
Comment (mezz) [reply mappings/net/minecraft/entity/Entity.mapping]: Going with `isPartOf`

Comment (asiekierka) [reply mappings/net/minecraft/entity/Entity.mapping]: isOwnerOf?

Comment (mezz) [reply mappings/net/minecraft/entity/Entity.mapping]: isOwnerOf would be the opposite logic

**Pull request merged**